tests/abstract_test.py

```diff
@@ -212,6 +251,32 @@ def test_abstract_the_dynamics():
 
 test_abstract_the_dynamics.slow = True
 
+def test_abstract_the_dynamics_dual():
+    """test_abstract_the_dynamics using dual-simulation algorithm"""
+    dom = pc.box2poly([[0.0, 10.0], [0.0, 20.0]])
+    ppp = define_partition(dom)
+    sys = define_dynamics(dom)
+    logger.info(sys)
+
+    disc_options = {'N':3, 'trans_length':2, 'min_cell_volume':1.5}
+
+    ab = abstract.discretize(ppp, sys, plotit=False,
+                             save_img=False, simu_type='dual',**disc_options)
+    assert(ab.ppp.compute_adj() )
```

Comment:
The outer parentheses can be omitted, because `assert` is a statement (not a function).